core/src/main/java/org/apache/iceberg/FindFiles.java

```diff
@@ -197,7 +197,7 @@ public Builder inPartitions(PartitionSpec spec, List<StructLike> partitions) {
       }
 
       // when snapshot is not null
-      CloseableIterable<ManifestEntry<DataFile>> entries = new ManifestGroup(ops.io(), snapshot.manifests())
+      CloseableIterable<ManifestEntry<DataFile>> entries = new ManifestGroup(ops.io(), snapshot.dataManifests())
```

Comment:
The contract for this class is to return matching data files, not splits. We use it in some cases to avoid `newScan` that will return either `FileScanTask` or `CombinedScanTask` and will emit a scan event. I think we may be able to remove it, or update it to return both data and delete files. For now, this uses `dataManifests` to mimic the current behavior.